### PR TITLE
Add oryp5

### PR DESCRIPTION
**Pull request by crawfxrd**
TODO:

- [x] Finish devicetree.cb
- [x] Fix rotated display of edk2 payload
  - Panel 3 was set to 180 degree rotation in VBT
- [x] Fix NVMe drive detection
  - Clock for RP#21 was not configured correctly
- [x] Fix Ethernet
- [x] Fix webcam detection
  - Fixed in EC
- [x] Smart AMP
- [ ] Testing
- [ ] Documentation

Ref: system76/firmware-open#113

### CODE REVIEW

**Comment by crawfxrd** [comment]
Errors from coreboot:

```
No more variable MTRRs: 10
No more variable MTRRs: 10
No more variable MTRRs: 10
Unable to locate Global NVS
Unable to locate Global NVS
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
ERROR: Unknown MCH (0x3ec4) in load_table
ERROR: Unknown MCH (0x3ec4) in load_table
ERROR: Unknown MCH (0x3ec4) in load_table
ERROR: Unknown MCH (0x3ec4) in load_table
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
gpio_pad_reset_config_override: Logical to Chipset mapping not found
MRC: Could not find region 'UNIFIED_MRC_CACHE'
system76: failed to find DGPU fn 4
system76: failed to find DGPU fn 5
system76: failed to find DGPU fn 6
system76: failed to find DGPU fn 7
I2C: 00:4e missing read_resources
```

**Comment by jackpot51** [comment]
system76: failed to find DGPU fn 4 and others like it are not errors. Most NVIDIA GPUs only have functions 0 to 3 implemented

**Comment by jackpot51** [comment]
gpio_pad_reset_config_override: Logical to Chipset mapping not found happens on most other boards. I haven't found the cause

**Comment by crawfxrd** [comment]
romstage MTRR issue only happens on cold boot, where it reports extra ranges.

```
top_of_ram = 0x79eff000
MTRR Range: Start=78eff000 End=78f00000 (Size 1000)
MTRR Range: Start=78f00000 End=79000000 (Size 100000)
MTRR Range: Start=79000000 End=79800000 (Size 800000)
MTRR Range: Start=79800000 End=79c00000 (Size 400000)
MTRR Range: Start=79c00000 End=79e00000 (Size 200000)
MTRR Range: Start=79e00000 End=79e80000 (Size 80000)
MTRR Range: Start=79e80000 End=79ec0000 (Size 40000)
MTRR Range: Start=79ec0000 End=79ee0000 (Size 20000)
MTRR Range: Start=79ee0000 End=79ef0000 (Size 10000)
MTRR Range: Start=79ef0000 End=79ef8000 (Size 8000)
No more variable MTRRs: 10
No more variable MTRRs: 10
No more variable MTRRs: 10
```
vs on reboot
```
top_of_ram = 0x79f00000
MTRR Range: Start=78f00000 End=79000000 (Size 100000)
MTRR Range: Start=79000000 End=79800000 (Size 800000)
MTRR Range: Start=79800000 End=79c00000 (Size 400000)
MTRR Range: Start=79c00000 End=79e00000 (Size 200000)
MTRR Range: Start=79e00000 End=79f00000 (Size 100000)
MTRR Range: Start=7a000000 End=7b000000 (Size 1000000)
MTRR Range: Start=ff000000 End=0 (Size 1000000)
```

File: src/mainboard/system76/oryp5/devicetree.cb [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	register "gpe0_dw1" = "PMC_GPP_G"
+	register "gpe0_dw2" = "PMC_GPP_E"
+
+# Actual device tree
+	device cpu_cluster 0 on
+		device lapic 0 on end
+	end
+
+	device domain 0 on
+		subsystemid 0x1558 0x95e6 inherit
```
Comment (crawfxrd) [inline src/mainboard/system76/oryp5/devicetree.cb:230]: This is overridden by `CONFIG_SUBSYSTEM_{DEVICE,VENDOR}_ID` from the firmware-open `coreboot.config` file.

File: src/mainboard/system76/oryp5/Kconfig [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	bool
+	default y
+
+config UART_FOR_CONSOLE
+	int
+	default 2
+
+config MAX_CPUS
+	int
+	default 12
```
Comment (crawfxrd) [inline src/mainboard/system76/oryp5/Kconfig:63]: ARK shows max 12 threads for the CPU, but we have 16 set for other boards.

**Comment by crawfxrd** [comment]
Audio output on Linux does not work; Works on Windows. (Headset prompt does open with subsystemid set to 1558:96e1.)

**Review by jackpot51 — APPROVED** [review]

**Pull request merged**